src/test/java/com/exasol/bucketfs/ReadEnabledBucketIT.java

```diff
@@ -42,28 +31,48 @@ void testGetDefaultBucket() {
 
     // [itest->dsn~bucket-lists-its-contents~2]
     @Test
-    void testListBucketContentsWithRootPath() throws BucketAccessException {
+    void listBucketContentsWithRootPath() throws BucketAccessException {
         assertThat(getDefaultBucket().listContents(), hasItem("EXAClusterOS/"));
     }
 
     // [itest->dsn~bucket-lists-its-contents~2]
-    @ValueSource(strings = { "EXAClusterOS/", "/EXAClusterOS/" })
+    @ValueSource(strings = { "/EXAClusterOS/", "EXAClusterOS/" })
     @ParameterizedTest
-    void testListBucketContents(final String pathInBucket) throws BucketAccessException {
+    void listContents(final String pathInBucket) throws BucketAccessException {
         assertThat(getDefaultBucket().listContents(pathInBucket), hasItem(startsWith("ScriptLanguages")));
     }
 
     @Test
-    void testListBucketContentsOfIllegalPathThrowsException() {
+    void listContentsWithWrongReadPassword_Fails() throws Exception {
+        final String bucketName = createBucket("dir/file.txt", "protected content");
+        final ReadOnlyBucket bucket = getBucket(bucketName, "wrong read password");
+        ExceptionAssertions.assertThrowsWithMessage(BucketAccessException.class, () -> bucket.listContents("dir/"), //
+                startsWith("E-BFSJ-3: Access denied trying to list "));
+    }
+
+    @Test
+    void downloadWithWrongReadPassword_Fails() throws Exception {
+        final String pathInBucket = "dir/file.txt";
+        final String bucketName = createBucket(pathInBucket, "protected content");
+        final ReadOnlyBucket bucket = getBucket(bucketName, "wrong read password");
+        ExceptionAssertions.assertThrowsWithMessage(BucketAccessException.class,
+                () -> bucket.downloadFileAsString(pathInBucket), //
+                startsWith("E-BFSJ-3: Access denied trying to download "));
+    }
+
+    @Test
+    void listBucketContentsOfIllegalPath_ThrowsException() {
```

Comment:
No Underscores in method names please.

```suggestion
    void testListingBucketContentsOfIllegalPathThrowsException() {
```

Reply:
Similar in other tests.